src/RequestModelFactory.php

```diff
@@ -44,7 +45,11 @@ private function processModel(ServerRequestInterface $request, RequestModelInter
         $requestData = $this->getRequestData($request);
         $model->setRequestData($requestData);
         if ($model instanceof ValidatableModelInterface) {
-            $this->validateRequest($model, $requestData);
+            $result = $this->createValidator($model)->validate($model);
+            $errors = $this->getErrorsFromValidationResult($result);
+            if (!empty($errors)) {
+                throw new RequestValidationException($errors);
+            }
```

Comment:
If `getErrorsFromValidationResult` will be moved to exception class it can looks like this:
```php
            $result = $this->createValidator($model)->validate($model);

            if (!$result->isValid()) {
                throw new RequestValidationException($result);
            }
```

Reply:
it is not empty when there are no errors

```
Array
(
    [body.login] => Yiisoft\Validator\Result Object
        (
            [errors:Yiisoft\Validator\Result:private] => Array
                (
                )

        )

    [body.password] => Yiisoft\Validator\Result Object
        (
            [errors:Yiisoft\Validator\Result:private] => Array
                (
                )

        )

)
```

Reply:
Is this a bug or feature?

Reply:
> Is this a bug or feature?

I think this is normal: success result is also result.

Reply:
I don't understand why empty results are needed in resultset.

Reply:
This is not empty results, this is success results.

Reply:
made through the generator